File: 3085-minimum-deletions-to-make-string-k-special/3085-minimum-deletions-to-make-string-k-special.cs
public class Solution {
    public int MinimumDeletions(string word, int k) {
        var freq = new int[26];
        foreach(var ch in word) freq[ch - 'a']++;

        var uniqFreq = new HashSet<int>(freq);

        var minRemove = int.MaxValue;
        // we assume each freq as smallest in answer, and then check which numbers need to be removed to make it true
        foreach(var minFreq in uniqFreq) {
            if (minFreq == 0) continue;

            int currRemove = 0;
            foreach(var f in freq) {
                if(f < minFreq) currRemove += f;
                else if (f > minFreq + k) currRemove += f - minFreq - k;
            }
            minRemove = Math.Min(minRemove, currRemove);
        }

        return minRemove;
    }
}

File: 3085-minimum-deletions-to-make-string-k-special/3085-minimum-deletions-to-make-string-k-special.java
class Solution {
    public int minimumDeletions(String word, int k) {
        var freq = new int[26];
        for(var ch : word.toCharArray()) freq[ch-'a']++;
        var uniqFreq = Arrays.stream(freq).boxed().collect(Collectors.toSet());

        var minRemove = Integer.MAX_VALUE;
        // we assume each freq as smallest in answer, and then check which numbers need to be removed to make it true
        for(var minFreq : uniqFreq){
            if(minFreq == 0) continue;
            var currRemove = 0;

            for(var f : freq){
                if(f < minFreq) currRemove += f;
                else if(f > minFreq + k) currRemove += f - minFreq - k; // remove occurences outside minFreq+k range
            }
            minRemove = Math.min(minRemove, currRemove);
        }
        
        return minRemove;
    }
}